integration-test/src/main/java/com/sequenceiq/it/cloudbreak/assertion/freeipa/FreeIpaUserSyncAssertion.java

```diff
@@ -0,0 +1,51 @@
+package com.sequenceiq.it.cloudbreak.assertion.freeipa;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+import com.sequenceiq.freeipa.api.v1.freeipa.user.model.FailureDetails;
+import com.sequenceiq.it.cloudbreak.FreeIpaClient;
+import com.sequenceiq.it.cloudbreak.assertion.Assertion;
+import com.sequenceiq.it.cloudbreak.dto.freeipa.FreeIpaUserSyncTestDto;
+
+public class FreeIpaUserSyncAssertion {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaUserSyncAssertion.class);
+
+    private FreeIpaUserSyncAssertion() {
+    }
+
+    public static Assertion<FreeIpaUserSyncTestDto, FreeIpaClient> userSyncTakesLessThanGivenDuration(long givenDuration) {
+        return (testContext, freeIpaUserSyncTestDto, freeIpaClient) -> {
+            long duration = freeIpaUserSyncTestDto.getResponse().getEndTime() - freeIpaUserSyncTestDto.getResponse().getStartTime();
+
+            LOGGER.info(String.format("FreeIpa last user sync duration took %d sec", TimeUnit.MILLISECONDS.toSeconds(duration)));
+            assertThat(duration)
```

Comment:
As I know, the exception is handled by our homework. The assert will throw an Error. We were an issue with this, the framework didn't handle the error correctly. The test finished there and validate nothing

Reply:
As far as I know: https://github.com/hortonworks/cloudbreak/blob/29c1eb460e221512c125cb9b6e09205ca7117046/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/context/TestContext.java#L318-L332
part should handle the failed assertions in the `then` steps.